src/guides/anatomy-of-an-http-transaction.md

```diff
@@ -0,0 +1,360 @@
+the API docs for each of those.
+
+## Create the Server
+
+Any node web server application will at some point have to create a web server
+object. This is done by using [`createServer`][].
+
+```javascript
+var http = require('http');
+
+var server = http.createServer(function(request, response) {
+  // magic happens here!
+});
+```
+
+The function that's passed in to [`createServer`][] is called once for every HTTP
+request that's made against that server, so it's called the request handler. In
+fact, the [`Server`][] object returned by [`createServer`][] is also an
+[`EventEmitter`][], and what we have here is just shorthand for creating a
+`server` object and then adding the listener later.
+
+```javascript
+var server = http.createServer();
+server.on('request', function(request, response) {
+  // the same kind of magic happens here!
+});
+```
+
+When an HTTP request hits the server, node calls the request handler function
+with a few handy object for dealing with the transaction, `request` and
+`response`. We'll get to those shortly.
+
+In order to have this thing actually serving requests, the [`listen`][] method
+needs to be called on the `server` object. In most cases, all you'll need to put
+for arguments is the port number you want the server to listen on, but there are
+some other options too, so consult the [API reference][].
+
+## Method, URL and Headers
+
+When handling a request, the first thing you'll probably want to do is look at
+the method and URL, so that appropriate actions can be taken. Node makes this
+relatively painless by putting handy properties onto the `request` object.
+
+```javascript
+var method = request.method;
+var url = request.url;
+```
+> **Note:** The `request` object is an instance of [`IncomingMessage`][].
+
+The `method` here will always be a normal HTTP method/verb. The `url` is the
+full URL without the server, protocol or port, so everything after and including
+the third forward slash in a well-formed URL.
+
+Headers are also not far away. They're in their own object on `request` called
+`headers`.
+
+```javascript
+var headers = request.headers;
+var userAgent = headers['user-agent'];
+```
+
+It's important to note here that all headers are represented in lower-case only,
+regardless of how the client actually sent them. This is helps to simplify the
+task of parsing headers for whatever purpose.
+
+## Request Body
+
+Many HTTP requests, like `POST` or `PUT` for example, might have a body as part
+of the request.
+
+To get at the body data, it's a little different from accessing headers. The
+`request` object that's passed in to a handler implements the
+[`ReadableStream`][] interface. This stream can be listened to or piped elswhere
+just like any other stream. We can grab the data right out of the stream by
+listening to the stream's `'data'` and `'end'` events.
+
+```javascript
+var body = '';
+request.on('data', function(chunk) {
+  body += chunk;
```

Comment:
(Also, it might be important to note that I'm not saying we should talk about the unicode problem in this doc, just that we should show the safe method in the examples without comment.)